plugins/woocommerce/includes/class-wc-payment-gateways.php

```diff
@@ -113,6 +117,153 @@ public function init() {
 		}
 
 		ksort( $this->payment_gateways );
+
+		add_action( 'wc_payment_gateways_initialized', array( $this, 'on_payment_gateways_initialized' ), 10, 1 );
```

Comment:
We could also adopt the private accessible method pattern here:

```suggestion
		self::add_action( 'wc_payment_gateways_initialized', array( $this, 'on_payment_gateways_initialized' ) );
```

This doesn't stop third parties from unhooking the callback, but it makes it clear `on_payment_gateways_initialized` is not intended for direct access. I also dropped the priority and number of args, since they are surplus/defaults.

Reply:
Yes, makes sense -- it's good to restrict the additions to the public API to the `wc_payment_gateways_initialized` hook. 💯